### PR TITLE
Hercules CI's effect to push to attic

### LINKED ISSUE #8: Hercules CI agent that pushes to an ad hoc Attic cache
Let's do this after it https://github.com/mlabs-haskell/cardano.nix/issues/3 and https://github.com/mlabs-haskell/cardano.nix/issues/2 since it should conceptually the same.

### LINKED ISSUE #17: Attic Binary Cache
Carried over from Milestone 1 #8

### PR DESCRIPTION
**Pull request by aciceri**
fixes #8 fixes #17

### CODE REVIEW

File: ci/default.nix [hunk 1]
```diff
@@ -1,8 +1,18 @@
 {inputs, ...}: {
   imports = [
     inputs.hercules-ci-effects.flakeModule
+    (import "${inputs.hercules-ci-effects}/effects/populate-cache/default.nix")
```
Comment (zmrocze) [inline ci/default.nix:4]: I wondered aboutt it: does it make a difference with or without `import`?

Comment (aciceri) [reply ci/default.nix:4]: No difference, it automatically imports it if you pass a string or a path. Don't know why I used `import` here.

**Comment by zmrocze** [comment]
I applied some fixes,
one of them is to make the hercules-ci-effects input point to my private repository. I don't have write access to our `mlabs-haskell/hercules-ci-effects`, but wanted this pr to already work, so that's why. Same goes for [mlabs-haskell/push-cache-effect](https://github.com/mlabs-haskell/push-cache-effect)

**Comment by brainrake** [comment]
Failing because of a bug. https://github.com/hercules-ci/hercules-ci-effects/pull/165#pullrequestreview-1852593841

**Comment by brainrake** [comment]
Assigning to Milestone 2 as Milestone 1 was accepted without this.

**Pull request merged**